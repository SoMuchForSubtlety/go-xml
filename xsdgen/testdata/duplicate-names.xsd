<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ns="http://example.org"
  targetNamespace="http://example.org">
  <xs:import namespace="http://example2.org" />
  <xs:element name="TypeA">
    <xs:complexType>
      <xs:sequence>
        <xs:element name="toplevel">
          <xs:simpleType>
            <xs:restriction base="xs:integer">
              <xs:totalDigits value="15" />
            </xs:restriction>
          </xs:simpleType>
        </xs:element>
      </xs:sequence>
    </xs:complexType>
  </xs:element>

  <xs:element name="Type1">
    <xs:complexType>
      <xs:sequence>
        <xs:element maxOccurs="1" minOccurs="0" name="TypeA" nillable="false" type="ns:TypeA">
        </xs:element>
      </xs:sequence>
    </xs:complexType>
  </xs:element>
</xs:schema>
