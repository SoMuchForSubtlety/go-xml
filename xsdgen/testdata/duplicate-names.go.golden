// Code generated by xsdgen.test. DO NOT EDIT.

package ws

type Type1 struct {
	TypeA TypeA `xml:"http://example.org TypeA,omitempty"`
}

type Type2 struct {
	TypeA []TypeA1 `xml:"http://example2.org TypeA"`
}

type TypeA struct {
	Toplevel int `xml:"http://example.org toplevel"`
}

type TypeA1 struct {
	Nested int `xml:"http://example2.org nested"`
}
